bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/types/StateDescriptionFragmentBuilder.java

```diff
@@ -0,0 +1,239 @@
+                        && options == null) {
+                    return null;
+                }
+                final Boolean ro = readOnly;
+                return new StateDescription(minimum, maximum, step, pattern, ro == null ? false : ro.booleanValue(),
+                        options);
+            }
+
+        };
+    }
+
+    /**
+     * Set the maximum for the resulting {@link StateDescriptionFragment}.
+     *
+     * @param maximum the maximum for the resulting {@link StateDescriptionFragment}.
+     * @return this builder.
+     */
+    public StateDescriptionFragmentBuilder withMaximum(BigDecimal maximum) {
+        this.maximum = maximum;
+        return this;
+    }
+
+    /**
+     * Set the minimum for the resulting {@link StateDescriptionFragment}.
+     *
+     * @param minimum the minimum for the resulting {@link StateDescriptionFragment}.
+     * @return this builder.
+     */
+    public StateDescriptionFragmentBuilder withMinimum(BigDecimal minimum) {
+        this.minimum = minimum;
+        return this;
+    }
+
+    /**
+     * Set the step for the resulting {@link StateDescriptionFragment}.
+     *
+     * @param step the step for the resulting {@link StateDescriptionFragment}.
+     * @return this builder.
+     */
+    public StateDescriptionFragmentBuilder withStep(BigDecimal step) {
+        this.step = step;
+        return this;
+    }
+
+    /**
+     * Set the pattern for the resulting {@link StateDescriptionFragment}.
+     *
+     * @param pattern the pattern for the resulting {@link StateDescriptionFragment}.
+     * @return this builder.
+     */
+    public StateDescriptionFragmentBuilder withPattern(String pattern) {
+        this.pattern = pattern;
+        return this;
+    }
+
+    /**
+     * Set readOnly for the resulting {@link StateDescriptionFragment}.
+     *
+     * @param readOnly readOnly for the resulting {@link StateDescriptionFragment}.
+     * @return this builder.
+     */
+    public StateDescriptionFragmentBuilder withReadOnly(Boolean readOnly) {
+        this.readOnly = readOnly;
+        return this;
+    }
+
+    /**
+     * Set the {@link StateOption}s for the resulting {@link StateDescriptionFragment}.
+     *
+     * @param options the {@link StateOption}s for the resulting {@link StateDescriptionFragment}.
+     * @return this builder.
+     */
+    public StateDescriptionFragmentBuilder withOptions(List<StateOption> options) {
+        this.options = options;
+        return this;
+    }
+
+    /**
+     * Merge the given {@link StateDescriptionFragment}. Set all unset ({@code null}) fields of this builder to the
+     * values
+     * from the
+     * given {@link StateDescriptionFragment}.
+     *
+     * @param fragment a {@link StateDescriptionFragment} this builder should merge in.
+     * @return the builder.
+     */
+    public StateDescriptionFragmentBuilder mergeStateDescriptionFragment(StateDescriptionFragment fragment) {
```

Comment:
IMHO this method mixes the to use-cases again: One is the "providing" side, i.e. a convenient API to express what aspects an actor "knows" about the state description. The other one is to then merge all this information in the StateDescriptionService - and this is internal logic and shouldn't be exposed to bindings et al.